src/ReactiveDomain.Foundation.Tests/when_serializing_with_json_message_serializer.cs

```diff
@@ -0,0 +1,211 @@
+            var deserialized = serializer.Deserialize(eventData);
+            Assert.IsType<TestObject2>(deserialized);
+            var testObject2 = (TestObject2)deserialized;
+            Assert.True(string.CompareOrdinal(testObject.Data2, testObject2.Data2) == 0);
+        }
+
+        [Fact]
+        public void can_deserialize_to_a_specified_type() {
+            var serializer = new JsonMessageSerializer();
+            var eventData = serializer.Serialize(testObject);
+            //n.b. testObject is type TestObject
+            var deserialized = serializer.Deserialize(eventData, typeof(TestObject2));
+            Assert.IsType<TestObject2>(deserialized);
+            var testObject2 = (TestObject2)deserialized;
+            Assert.True(string.CompareOrdinal(testObject.Data2, testObject2.Data2) == 0);
+
+            var newTestObject2 = serializer.Deserialize<TestObject2>(eventData);
+            Assert.True(string.CompareOrdinal(testObject.Data2, newTestObject2.Data2) == 0);
+        }
+
+        [Fact]
+        public void can_throw_if_type_not_found() {
+            var serializer = new JsonMessageSerializer();
+
+            //n.b. setting header to non existent assembly
+            var headerName = serializer.EventClrQualifiedTypeHeader;
+            var headerData = $"{typeof(TestObject2).FullName},dne-assembly";
+            var headers = new Dictionary<string, object> {{headerName, headerData}};
+            var eventData = serializer.Serialize(testObject, headers);
+            //confirm type not found
+            var deserialized = serializer.Deserialize(eventData);
+            Assert.IsType<JObject>(deserialized);
+            //request throw on type not found
+            serializer.ThrowOnTypeNotFound = true;
+            Assert.Throws<InvalidOperationException>(() => serializer.Deserialize(eventData));
+        }
+
+        [Fact]
+        public void can_override_target_assembly() {
+            var serializer = new JsonMessageSerializer();
+
+            //n.b. setting header to non existent assembly
+            var headerName = serializer.EventClrQualifiedTypeHeader;
+            var headerData = $"{typeof(TestObject2).FullName},dne-assembly";
+            var headers = new Dictionary<string, object> { { headerName, headerData } };
+            var eventData = serializer.Serialize(testObject, headers);
+            //confirm type not found
+            var deserialized = serializer.Deserialize(eventData);
+            Assert.IsType<JObject>(deserialized);
+            //override assembly
+            serializer.AssemblyOverride = Assembly.GetExecutingAssembly();
+            deserialized = serializer.Deserialize(eventData);
+            var testObject2 = (TestObject2)deserialized;
+            Assert.True(string.CompareOrdinal(testObject.Data2, testObject2.Data2) == 0);
+        }
+
+        [Fact]
+        public void can_deserialize_from_either_fully_or_qualified_qualified_names() {
```

Comment:
This test name doesn't make sense. Perhaps `can_deserialize_from_either_fully_or_partially_qualified_names`?

Reply:
yes